File: Data Structures & Algorithms/binary-search/submission-2.java

class Solution{
  public int search(int[] nums, int target) {

    int n=nums.length;
    // Repeat until the pointers low and high meet each other
    int low=0;
    int high=n-1;
    while (low <= high) {
      int mid = low + (high - low) / 2;

      if (target == nums[mid])
        return mid;

      if (target > nums[mid])
        low = mid + 1;

      else
        high = mid - 1;
    }

    return -1;
  }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-2.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        l = nums[0]
        r = nums[0]

        while True:
            l = nums[l]
            r = nums[nums[r]]

            if l == r:
                break

        l = nums[0]

        while l != r:
            l = nums[l]
            r = nums[r]

        return l